### PR TITLE
Put and patch

### LINKED ISSUE #21: Add possibility to create PUT and PATCH requests to object recourse uris
#### Proposed solution
It should be possible to update metadata objects fully or partially. Make it possible to update objects via /objects/<schema>/<accessionId> and via /<schema/<accessionId> (redirection). 

#### DoD (Definition of Done)
- Objects can be updates via endpoints

#### Testing
- Test with unit tests

### PR DESCRIPTION
**Pull request by blankdots**
### Description

support `put` and `patch` operations in drafts

### Related issues

Fixes #21 

### Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)

### Changes Made

1. Add feature to PUT XML in drafts and save it the back-up.
2. Add feature for PATCH JSON.
   * current only supported in drafts
   * raise 415 for XML
3. Add tests for features
4. other small fixes

### Testing

- [x] Unit Tests
- [x] Integration Tests
- [ ] Tests do not apply
- [ ] Needs testing (start an issue or do a follow up PR about it)

### Mentions

we could also add them to `objects` but was not sure, @otahontas and @csc-jm what do you think ?

### CODE REVIEW

File: metadata_backend/api/operators.py [hunk 1]
```diff
@@ -361,6 +427,25 @@ async def _format_data_to_replace_and_add_to_db(self, schema_type: str,
         return await self._replace_object_from_db(schema_type, accession_id,
                                                   data)
 
+    async def _format_data_to_update_and_add_to_db(self, schema_type: str,
+                                                   accession_id: str,
+                                                   data: Any) -> str:
+        """Format and update data in database.
+
+        Must be implemented by subclass.
+        """
```
Comment (csc-jm) [inline metadata_backend/api/operators.py:436]: This was missing:
```suggestion
        """Format and update data in database.

        :param schema_type: Schema type of the object to replace.
        :param accession_id: Identifier of object to replace.
        :param data: Metadata object
        :returns: Accession Id for object inserted to database
        """
```

File: metadata_backend/api/operators.py [hunk 2]
```diff
@@ -457,14 +545,42 @@ async def _format_data_to_replace_and_add_to_db(self, schema_type: str,
         """Format XML metadata object and add it to db.
 
         XML is validated, then parsed to json and json is added to database.
-        After successful json insertion, xml itself is backed up to database.
+        XML .
```
Comment (csc-jm) [inline metadata_backend/api/operators.py:548]: ?

File: metadata_backend/api/operators.py [hunk 3]
```diff
@@ -457,14 +545,42 @@ async def _format_data_to_replace_and_add_to_db(self, schema_type: str,
...
+
+        Patch update for XML not supported
+
+        :param schema_type: Schema type of the object to replace.
+        :param accession_id: Identifier of object to replace.
+        :param data: Original xml content
+        :raises: NotImplemeted
+        """
+        reason = "XML patching is not possible."
+        raise web.HTTPUnsupportedMediaType(reason=reason)
```
Comment (csc-jm) [inline metadata_backend/api/operators.py:583]: Does not raise `NotImplemented`

Comment (blankdots) [reply metadata_backend/api/operators.py:583]: good catch

File: metadata_backend/api/operators.py [hunk 4]
```diff
@@ -457,14 +545,42 @@ async def _format_data_to_replace_and_add_to_db(self, schema_type: str,
...
+                                                   accession_id: str,
+                                                   data: str) -> str:
+        """Raise not implemented.
+
+        Patch update for XML not supported
+
+        :param schema_type: Schema type of the object to replace.
+        :param accession_id: Identifier of object to replace.
+        :param data: Original xml content
+        :raises: NotImplemeted
```
Comment (blankdots) [inline metadata_backend/api/operators.py:580]:
```suggestion
        :raises: HTTPUnsupportedMediaType
```

File: metadata_backend/api/operators.py (hunk 2 above)
Comment (blankdots) [reply metadata_backend/api/operators.py:548]: my bad :)

Comment (blankdots) [inline metadata_backend/api/operators.py:548]:
```suggestion
         After successful json insertion, xml itself is backed up to database
```

**Pull request merged**